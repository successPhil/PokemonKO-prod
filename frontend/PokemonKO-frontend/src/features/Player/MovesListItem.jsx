import TrainerContext from "../../contexts/TrainerContext";
import { useContext } from "react";
import { capitalizeFirst } from "../Enemy/EnemyData";
export default function MovesListItem({move, trainerTurn, toggleMenu, openMoves, selectPokemon, enemyPokemon, playerAttack}){
    const upperMove = capitalizeFirst(move.name)
    const { typeToClassname, typeToIcon} = useContext(TrainerContext);
    const typeClassName = typeToClassname[move.type];
    const icon = typeToIcon[move.type];
    const iconClassName = 'pokemon-type-icon'

    
    const handleClickMove = (move) => {
        if (trainerTurn){
            playerAttack(selectPokemon, move, enemyPokemon)
            toggleMenu(openMoves)
           
        }
    }

    return (
    <>
       <div className='gameboy-move-box' onClick={()=> handleClickMove(move)}>
        <div className='move-content' >
       <div className={`${iconClassName} ${typeClassName}`} >{icon}</div><div>{upperMove}</div>
        </div>
      </div>
    </>)

}