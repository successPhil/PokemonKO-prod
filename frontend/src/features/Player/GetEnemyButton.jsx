export default function GetEnemyButton () {
    return (
        <>
        <div className='get-enemy-container'>
            <div className='gameboy-menu-box'>
                <div className='get-enemy-content'>Fight!</div>
      </div>
    </div>
        </>
    )
}