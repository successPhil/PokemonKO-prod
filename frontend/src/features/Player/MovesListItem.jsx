import { capitalizeFirst } from "../Enemy/EnemyData";
import { typeToClassname } from "../../constants/typeToClassname";
import { typeToIcon } from "../../constants/typeToIcon";
export default function MovesListItem({move, trainerTurn, toggleMenu, openMoves, selectPokemon, enemyPokemon, playerAttack}){
    const upperMove = capitalizeFirst(move.name)
  
    const typeClassName = typeToClassname[move.type];
    const icon = typeToIcon[move.type];
    const iconClassName = 'pokemon-type-icon'

    
    const handleClickMove = (move) => {
        if (trainerTurn){
            playerAttack(selectPokemon, move, enemyPokemon)
            toggleMenu(openMoves)
           
        }
    }

    return (
    <>
       <div className='gameboy-move-box' onClick={()=> handleClickMove(move)}>
        <div className='move-content' >
       <div className={`${iconClassName} ${typeClassName}`} >{icon}</div><div>{upperMove}</div>
        </div>
      </div>
    </>)

}