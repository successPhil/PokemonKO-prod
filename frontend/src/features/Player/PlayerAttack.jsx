import { motion } from "framer-motion"
import TrainerContext from "../../contexts/TrainerContext"
import { useContext } from "react"

export default function PlayerAttack() {
    const { animateSelectAttack , animateColor, selectPokemon} = useContext(TrainerContext)

    const pokeTypes = selectPokemon.types.split(", ")

    return (
        <motion.div
        className="player-attack"
        initial={{x:800, y:170}}
        animate={animateSelectAttack ? { x: 1200, y: 0, rotate: 0 } : { x: 800, y: 170, rotate: 0 }}
        transition={{ duration: 0.4 }}
        >
            <div>
        <span className={`pokeball-attack ${animateColor}`}>Z</span>
        <span className={`pokeball-attack ${pokeTypes[0]}`} >u</span>
      </div>
        </motion.div>
    )
}