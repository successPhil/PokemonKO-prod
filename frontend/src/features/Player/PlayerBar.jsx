import React from 'react';

export default function PlayerBar({ isPlayer }) {
  const containerClass = isPlayer ? 'player-bar' : 'enemy-bar';

  return (
    <div className={containerClass}>
      <div className='horizontal-bar'></div>
      <div className='vertical-bar'></div>
    </div>
  );
}
