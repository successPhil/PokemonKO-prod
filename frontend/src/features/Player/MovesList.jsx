import Pokeball from "../../styles/Pokeball";
import MovesListItem from "./MovesListItem";
export default function MovesList({moves, endTrainerTurn, trainerTurn, toggleMenu, openMoves, selectPokemon, trainerDialogue, setTrainerDialogue, enemyPokemon, playerAttack}) {
    return (<div className="gameboy-moves-container">
    <div className='gameboy-move-container'>
        {moves.map((move) => (<MovesListItem move={move} key={move.id} endTrainerTurn={endTrainerTurn} trainerTurn={trainerTurn} toggleMenu={toggleMenu} openMoves={openMoves} selectPokemon={selectPokemon} trainerDialogue={trainerDialogue} setTrainerDialogue={setTrainerDialogue} enemyPokemon={enemyPokemon} playerAttack={playerAttack} />))}
    </div>
    <div className="top-left-menu">
        <Pokeball />
      </div>
      <div className="top-right-menu">
        <Pokeball />
      </div>
      <div className="bottom-left-menu">
        <Pokeball />
      </div>
      <div className="bottom-right-menu">
        <Pokeball />
      </div>

    </div>)
}

