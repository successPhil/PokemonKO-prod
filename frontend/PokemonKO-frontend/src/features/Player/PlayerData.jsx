import GameText from "../../styles/PokemonGB"
import ProgressBar from "../ProgressBar/ProgressBar"
import PlayerBar from "./PlayerBar"
import { capitalizeFirst } from "../Enemy/EnemyData";
import TrainerContext from "../../contexts/TrainerContext";
import { useContext } from "react";
import SelectImage from "./SelectImage";
import PlayerAttack from "./PlayerAttack";

export default function PlayerData(){
    const { selectPokemon, animateSelectAttack } = useContext(TrainerContext)
    const isPlayer = true;
    if (selectPokemon){
        return(<>
        <SelectImage selectImage={selectPokemon.back_image_url}/>
        {animateSelectAttack && <PlayerAttack />}
        <PlayerBar isPlayer={isPlayer}/>
        <div className="player-name">
        <GameText>{capitalizeFirst(selectPokemon.name)}</GameText>
        </div>
        <div className="player-level">
            <GameText>LVL:{selectPokemon.level}</GameText>
        </div>
        <div className="player-health-bar">
        <GameText>HP: </GameText>
        <ProgressBar  value={selectPokemon.health} maxValue={selectPokemon.max_health} barColor={'#DA2C38'} />
        </div>
        <div className="player-xp-bar">
        <GameText>XP:</GameText>
        <ProgressBar value={selectPokemon.experience} maxValue={selectPokemon.totalXP} barColor={'#D5E5AE'}/>
        </div>
        </>) 
    }
}