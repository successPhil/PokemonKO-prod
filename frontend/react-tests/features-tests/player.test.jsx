import GetEnemyButton from '../../src/features/Player/GetEnemyButton'
import MovesList from '../../src/features/Player/MovesList'
import MovesListItem from '../../src/features/Player/MovesListItem'
import PlayerAttack from '../../src/features/Player/PlayerAttack'
import PlayerBar from '../../src/features/Player/PlayerBar'
import PlayerData from '../../src/features/Player/PlayerData'
import SelectImage from '../../src/features/Player/SelectImage'

import { expect, test } from 'vitest'

test("Player feature components render without crashing", ()=>{
    const getEnemyButton = <GetEnemyButton />
    expect(getEnemyButton).toMatchSnapshot()
    const movesList = <MovesList/>
    expect(movesList).toMatchSnapshot()
    const movesListItem = <MovesListItem />
    expect(movesListItem).toMatchSnapshot()
    const playerAttack = <PlayerAttack />
    expect(playerAttack).toMatchSnapshot()
    const playerBar = <PlayerBar />
    expect(playerBar).toMatchSnapshot()
    const playerData = <PlayerData />
    expect(playerData).toMatchSnapshot()
    const selectImage = <SelectImage />
    expect(selectImage).toMatchSnapshot()
} )