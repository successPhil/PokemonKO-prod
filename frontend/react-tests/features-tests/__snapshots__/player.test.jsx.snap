// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Player feature components render without crashing 1`] = `<GetEnemyButton />`;

exports[`Player feature components render without crashing 2`] = `<MovesList />`;

exports[`Player feature components render without crashing 3`] = `<MovesListItem />`;

exports[`Player feature components render without crashing 4`] = `<PlayerAttack />`;

exports[`Player feature components render without crashing 5`] = `<PlayerBar />`;

exports[`Player feature components render without crashing 6`] = `<PlayerData />`;

exports[`Player feature components render without crashing 7`] = `<SelectImage />`;
